KAADO-44: Create edit card page

tsoibet opened the pull request:
- Create edit card page
- Fix updateCard bug

tsoibet committed:
KAADO-44:
- Create edit card page
- Fix updateCard bug

nasudon commented on src/app/cards/edit/[id]/Form.tsx:
```diff
@@ -0,0 +1,242 @@
+    function handleImgChange(event: ChangeEvent<HTMLInputElement>) {
+        const { name } = event.target;
+        const file = event.target.files && event.target.files[0];
+        console.log(file);
```
Do we need to log this?

tsoibet replied:
Fixed

nasudon commented on src/app/cards/edit/[id]/Form.tsx:
```diff
@@ -0,0 +1,242 @@
+    async function saveBtnClickHandler(event: React.FormEvent) {
+        event.preventDefault();
+        setLoading(true);
+        console.log('clicked');
```
Do we need to log this?

tsoibet replied:
Fixed

nasudon commented on src/app/cards/edit/[id]/Form.tsx:
```diff
@@ -0,0 +1,242 @@
+                        onChange={(event) => handleImgChange(event)}
+                        disabled={loading}
+                        className="h-0"
+                    />
```
I think we should "hide" the default input

tsoibet replied:
Fixed

nasudon commented on src/app/cards/edit/[id]/Form.tsx:
```diff
@@ -0,0 +1,242 @@
+                        onChange={(event) => handleImgChange(event)}
+                        disabled={loading}
+                        className="h-0"
+                    />
```
Same here. I think we should "hide" the default input

tsoibet replied:
Fixed

nasudon commented on src/app/cards/edit/[id]/Form.tsx:
```diff
@@ -0,0 +1,242 @@
+                        />
+                    </label>
+                </div>
+
```
[NIT] Unused new line

tsoibet replied:
Fixed

nasudon commented on src/app/cards/edit/[id]/Form.tsx:
```diff
@@ -0,0 +1,242 @@
+                        id="name"
+                        type="text"
+                        value={card.name}
+                        onChange={(event) => setCard({ ...card, name: event.target.value })}
```
[NIT] The `(event) => setCard({ ...card, name: event.target.value })` repeated many times in this file. Shall we move it out as a function?

tsoibet replied:
Fixed

nasudon commented on src/app/cards/edit/[id]/page.tsx:
```diff
@@ -0,0 +1,50 @@
+            note,
+        });
+        if (updatedCard) {
+            revalidatePath('/');
```
Shall we revalidate the `/cards/detail/${params.id}` as well?

tsoibet replied:
Fixed.
Added `/cards/${params.id}` too.

nasudon reviewed (changes requested):
Please check my comments. Some of them are just nitpick and please focus on the UI part

tsoibet committed:
KAADO-44:
- Fix file input style for safari
- Revalidate all related path after updating card
- Refactor code

tsoibet committed:
KAADO-44:
- Format code

nasudon reviewed (approved):
LGTM
